build/riscv64-task-link.x

```diff
@@ -0,0 +1,91 @@
+INCLUDE memory.x
+
+PROVIDE(_heap_size = 0);
+
+ENTRY(_start);
+
+SECTIONS
+{
+  PROVIDE(_stack_start = ORIGIN(STACK) + LENGTH(STACK));
+
+  /* ### .text */
+  .text : {
+    _stext = .;
+    *(.text.start*); /* try and pull start symbol to beginning */
+    *(.text .text.*);
+    . = ALIGN(8);
+    __etext = .;
+  } > RAM =0xdededede
+
+  /* ### .rodata */
+  .rodata : ALIGN(8)
+  {
+    *(.rodata .rodata.*);
+
+    /* 8-byte align the end (VMA) of this section.
+       This is required by LLD to ensure the LMA of the following .data
+       section will have the correct alignment. */
+    . = ALIGN(8);
+    __erodata = .;
+  } > RAM
+
+  /*
+   * Sections in RAM
+   *
+   * NOTE: the userlib runtime assumes that these sections
+   * are 8-byte aligned and padded to 8-byte boundaries.
+   */
+  .data : ALIGN(8) {
+    . = ALIGN(8);
+    __sdata = .;
+    *(.data .data.*);
```

Comment:
It looks like this script was from before we added `sdata` back, so non-zero static globals will still be put at their intended address. Adding `    *(.sdata .sdata.*);` should fix it.

Reply:
Done. Pushed a follow up commit.